src/scripts/hubot-jenkins-notifier.coffee

```diff
@@ -117,22 +136,46 @@ class JenkinsNotifier
     if !data || typeof data.build != 'object'
       @error new Error("Unable to process data"), req.body
       return
+
+    if query.trace
+      console.log data.build
+
     fullurl = data.build.full_url || data.build.url
 
-    if data.build.phase in ['FINISHED', 'FINALIZED']
+    if data.build.phase in ['COMPLETED']
```

Comment:
The FINIHSED phase does not exist according to toe notification plugin. Valid settings are "Job Started", "Job Completed", "Job Finalized" or "All Events", they map to STARTED, COMPLETED and FINALIZED.
The STARTED is now also handled and sends a message to the room, if wanted.

I wanted the user to be able to leave (the default option, which ist "All events".
In that case COMPLETED and FINALIZED are sent. If both are handled then the mechanism to remember the last result was broken. 
So I decided to not handle COMPLETED at all.

So, first COMPLETED notifications are ignored, leaving STARTED and FINALIZED.
Then first STARTED is handled, after that the rest of the notifications.

But you are right, it could be handled better and more clearer. I will adjust the code.

Do you think it is necessary to still handle FINISHED since it is not supported in the plugin anymore?